lib/msf/core/post/windows/mssql.rb

```diff
@@ -109,13 +109,27 @@ def check_sqlcmd
         # @param [String] query the query to execute
         # @param [String] instance the SQL instance to target
         # @param [String] server the SQL server to target
+        # @param [String] database the database to connect to upon successfully authenticating
+        # @param [String] username the username to authenticate as
+        # @param [String] password the password to authenticate with
         # @return [String] the result of query
-        def run_sql(query, instance = nil, server = '.')
+        def run_sql(query, instance = nil, server = '.', database: nil, username: nil, password: nil)
           target = server
           if instance && instance.downcase != 'mssqlserver'
             target = "#{server}\\#{instance}"
           end
-          cmd = "#{@sql_client} -E -S #{target} -Q \"#{query}\" -h-1 -w 200"
+          cmd = "#{@sql_client}"
+          cmd += " -d #{database}" unless database.nil? || database.empty?
```

Comment:
You can actually combine this using the `#blank?` method provided by rails.
```suggestion
          cmd += " -d #{database}" unless database.blank?
```

Reply:
Not so sure about the `blank?` and `present?` methods as in my testing they don't appear to exist on a `nil` object:

```
3.0.2 :039 > nil.blank?
(irb):39:in `<main>': undefined method `blank?' for nil:NilClass (NoMethodError)
        from /home/gwillcox/.rvm/gems/ruby-3.0.2@metasploit-framework/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from /home/gwillcox/.rvm/gems/ruby-3.0.2@metasploit-framework/bin/irb:23:in `load'
        from /home/gwillcox/.rvm/gems/ruby-3.0.2@metasploit-framework/bin/irb:23:in `<main>'
3.0.2 :040 > nil.nil?
 => true 
3.0.2 :041 > nil.present?
(irb):41:in `<main>': undefined method `present?' for nil:NilClass (NoMethodError)
        from /home/gwillcox/.rvm/gems/ruby-3.0.2@metasploit-framework/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from /home/gwillcox/.rvm/gems/ruby-3.0.2@metasploit-framework/bin/irb:23:in `load'
        from /home/gwillcox/.rvm/gems/ruby-3.0.2@metasploit-framework/bin/irb:23:in `<main>'
3.0.2 :042 > 
```